apps/search/src/components/AlgoliaConfigurationForm.tsx

```diff
@@ -0,0 +1,174 @@
+import { useAuthenticatedFetch } from "@saleor/app-sdk/app-bridge";
+
+import { Controller, useForm } from "react-hook-form";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { AlgoliaConfigurationFields } from "../lib/algolia/types";
+import { fetchConfiguration } from "../lib/configuration";
+import { Box, Button, Divider, Input, Text } from "@saleor/macaw-ui/next";
+import { useDashboardNotification } from "@saleor/apps-shared";
+import { AlgoliaSearchProvider } from "../lib/algolia/algoliaSearchProvider";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+import { useState } from "react";
+
+const formSchema = z.object({
```

Comment:
Schema probably should be somehow connected to the endpoint, to be sure both places expect the same shape of the data